packages/back-end/src/models/ProjectModel.ts

```diff
@@ -1,126 +1,82 @@
-import mongoose from "mongoose";
-import uniqid from "uniqid";
 import { DEFAULT_STATS_ENGINE } from "shared/constants";
-import { omit } from "lodash";
 import { hasReadAccess } from "shared/permissions";
+import { z } from "zod";
 import { ApiProject } from "../../types/openapi";
 import { ProjectInterface, ProjectSettings } from "../../types/project";
-import { ReqContext } from "../../types/organization";
-import { ApiReqContext } from "../../types/api";
+import { StatsEngine } from "../../types/stats";
+import { tuple } from "../util/types";
+import { baseSchema, MakeModelClass, Context } from "./BaseModel";
 
-const projectSchema = new mongoose.Schema({
-  id: {
-    type: String,
-    unique: true,
-  },
-  organization: {
-    type: String,
-    index: true,
+const engineTypes = tuple<StatsEngine>()(["bayesian", "frequentist"]);
```

Comment:
Why is this needed?  Can't we just use a const array?

Reply:
We can. I am just trying to have a safety check that the array items are kept in sync with `StatsEngine`. This is another instance where `StatsEngine` cannot be defined using the usual:

```typescript
const values = [...] as const;
type Values = typeof values[number];
```

Because the API in `back-end/types` is types only.

Reply:
I don't think this tuples approach is very readable.  I'd prefer if we either:

1. Use `[...] as const` like in your comment.  Can just define in a separate `.ts` file and import it to the `.d.ts` types file to export the type
2. Keep the type def as-is and change the line here to: `const engineTypes: StatsEngine[] = ["bayesian", "frequentist"]`.  I know it's not 100% the same, but we're not going to be adding new stats engines very often (if at all), so I'm not too worried about maintenance here.

Reply:
I got rid of it. I agree that it's not super intuitive but I'm not sure we can do better. `/types` is purposedly setup to prevent import of `.ts` files (`rootDir` is set to `/src`) which is good and sub-typing to `StatsEngine[]` breaks the assumptions needed for zod to make it a tuple in its validator.

Reply:
My bad. Locating the definitions under `./src` made it work.